cmd/buildah/rmi.go

```diff
@@ -160,17 +164,24 @@ func deleteImages(ctx context.Context, systemContext *types.SystemContext, store
 				// because only a copy of the image state is returned
 				image, err = getImage(ctx, systemContext, image.ID, store)
 				if err != nil || image == nil {
-					return errors.Wrapf(err, "error getting image after untag %q", image.ID)
+					if lastError != nil {
+						fmt.Fprintln(os.Stderr, lastError)
+					}
+					lastError = errors.Wrapf(err, "error getting image after untag %q", image.ID)
 				}
 			}
 
 			isParent, err := imageIsParent(store, image.TopLayer)
 			if err != nil {
-				return err
+				if lastError != nil {
+					fmt.Fprintln(os.Stderr, lastError)
+				}
+				lastError = errors.Wrapf(err, "error judging image %q", image.ID)
+				continue
 			}
 			// If the --all flag is not set and the image has named references or is
-			// a parent, do not elete image.
-			if len(image.Names) > 0 || (isParent && len(image.Names) == 0) && !removeAll {
+			// a parent, do not delete image.
+			if len(image.Names) > 0 && !removeAll {
```

Comment:
We need the `|| (isParent && len(image.Names) == 0)` condition to ensure that we don't delete an image that doesn't have a reference but is a parent for another image. Please see the example below:
```
➜  buildah git:(master) ✗ sudo buildah bud --layers -t image1 .
STEP 1: FROM alpine
Getting image source signatures
Copying blob sha256:8e3ba11ec2a2b39ab372c60c16b421536e50e5ce64a0bc81765c2e38381bcff6
 2.10 MiB / 2.10 MiB [======================================================] 0s
Copying config sha256:11cd0b38bc3ceb958ffb2f9bd70be3fb317ce7d255c8a4c3f4af30e298aa1aab
 1.48 KiB / 1.48 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: RUN mkdir hello
--> 88ab4f7e555f1cfa397292ffd9260768bc074f12c3c519e19ee00a4ae90dc465
STEP 3: FROM 88ab4f7e555f1cfa397292ffd9260768bc074f12c3c519e19ee00a4ae90dc465
STEP 4: ENV foo=bar
--> fc682a481b02988ff8ec6381400e2384835b447f7eddc42724f422eaf14a94b7
STEP 5: FROM fc682a481b02988ff8ec6381400e2384835b447f7eddc42724f422eaf14a94b7
STEP 6: RUN touch blah
--> c7dc1bda14ea433d45e843c8ddb1e3728e2e33de02cae1cafaa75113d153bc53
STEP 7: COMMIT image1
➜  buildah git:(master) ✗ sudo buildah images -a
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
11cd0b38bc3c         docker.io/library/alpine:latest                          Jul 6, 2018 10:14      4.67 MB
88ab4f7e555f         <none>                                                   Aug 13, 2018 10:53     4.68 MB
fc682a481b02         <none>                                                   Aug 13, 2018 10:53     4.68 MB
c7dc1bda14ea         localhost/image1:latest                                  Aug 13, 2018 10:53     4.68 MB
➜  buildah git:(master) ✗ sudo buildah bud --layers -t image2 .
STEP 1: FROM alpine
STEP 2: RUN mkdir hello
--> Using cache 88ab4f7e555f1cfa397292ffd9260768bc074f12c3c519e19ee00a4ae90dc465
STEP 3: FROM 88ab4f7e555f1cfa397292ffd9260768bc074f12c3c519e19ee00a4ae90dc465
STEP 4: RUN touch blah
--> d78e8e7fe86a2c954761f31201c07619d4a9cd03228d2a2d20f7b30cea49e5af
STEP 5: COMMIT image2
➜  buildah git:(master) ✗ sudo buildah images -a               
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
11cd0b38bc3c         docker.io/library/alpine:latest                          Jul 6, 2018 10:14      4.67 MB
88ab4f7e555f         <none>                                                   Aug 13, 2018 10:53     4.68 MB
fc682a481b02         <none>                                                   Aug 13, 2018 10:53     4.68 MB
c7dc1bda14ea         localhost/image1:latest                                  Aug 13, 2018 10:53     4.68 MB
d78e8e7fe86a         localhost/image2:latest                                  Aug 13, 2018 10:53     4.68 MB
➜  buildah git:(master) ✗ sudo buildah rmi alpine
untagged: docker.io/library/alpine:latest
➜  buildah git:(master) ✗ sudo buildah images -a
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
11cd0b38bc3c         <none>                                                   Jul 6, 2018 10:14      4.67 MB
88ab4f7e555f         <none>                                                   Aug 13, 2018 10:53     4.68 MB
fc682a481b02         <none>                                                   Aug 13, 2018 10:53     4.68 MB
c7dc1bda14ea         localhost/image1:latest                                  Aug 13, 2018 10:53     4.68 MB
d78e8e7fe86a         localhost/image2:latest                                  Aug 13, 2018 10:53     4.68 MB
➜  buildah git:(master) ✗ sudo buildah rmi image1
untagged: localhost/image1:latest
fc682a481b02988ff8ec6381400e2384835b447f7eddc42724f422eaf14a94b7

➜  buildah git:(master) ✗ sudo buildah images -a
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
11cd0b38bc3c         <none>                                                   Jul 6, 2018 10:14      4.67 MB
88ab4f7e555f         <none>                                                   Aug 13, 2018 10:53     4.68 MB
d78e8e7fe86a         localhost/image2:latest                                  Aug 13, 2018 10:53     4.68 MB
```

Reply:
Instead of deleting this code, the same validation code is shown below. If not modified, the following error will not be executed.
```
                        if isParent && len(image.Names) == 0 && !removeAll {
                                if lastError != nil {
                                        fmt.Fprintln(os.Stderr, lastError)
                                }
                                lastError = errors.Errorf("unable to delete %q (cannot be forced) - image has dependent child images", image.ID)
                                continue
                        }
```
The following also occurs: an error occurred while deleting image, but there is no error message because the code above cannot be executed.
```
➜  buildah git:(rmi-fix) ✗ sudo buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
e1ddd7948a1c         docker.io/library/busybox:latest                         Aug 1, 2018 06:20      1.38 MB
e1ddd7948a1c         localhost/zz:latest                                      Aug 1, 2018 06:20      1.38 MB
b32225dc1e44         <none>                                                   Aug 10, 2018 15:32     1.38 MB
➜  buildah git:(rmi-fix) ✗ sudo buildah rmi -f e1dd
➜  buildah git:(rmi-fix) ✗ sudo buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
e1ddd7948a1c         docker.io/library/busybox:latest                         Aug 1, 2018 06:20      1.38 MB
e1ddd7948a1c         localhost/zz:latest                                      Aug 1, 2018 06:20      1.38 MB
b32225dc1e44         <none>                                                   Aug 10, 2018 15:32     1.38 MB
➜  buildah git:(rmi-fix) ✗ sudo ./buildah rmi -f e1dd
unable to delete "e1ddd7948a1c31709a23cc5b7dfe96e55fc364f90e1cebcde0773a1b5a30dcda" (cannot be forced) - image has dependent child images
```